src/test/java/com/netcracker/study/objects/borders/WallTest.java

```diff
@@ -0,0 +1,16 @@
+package com.netcracker.study.objects.borders;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class WallTest {
+
+    @Test
+    public void breakAndShootTest(){
+        Wall testWall = new Wall(1);
+
+        testWall.onGrenadeHit();
```

Comment:
you may return boolean type on testWall.onGrenadeHit(); showing either the object is alive or refactor testWall.doesStopBullet() method to testWall.isAlive()

Reply:
I've get it... testWall.doesStopBullet() -> testWall.isImmutable()